Sources/Subs-Db-mysql.php

```diff
@@ -749,6 +751,21 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $dis
 		),
 		$connection
 	);
+	
+	if(!empty($keys) && (count($keys) > 0) && $method == '' && $returnmode > 0)
+	{
+		if ($returnmode == 1)
```

Comment:
With switch your mean case?
Why should be a case here better as if?
use more space/loc and we got only two possibilities